File: 0941-valid-mountain-array/0941-valid-mountain-array.ts
function validMountainArray(arr: number[]): boolean {
    //initial condition
    if(arr.length < 3) {
        return false;
    }
    
    let i = 0;
    
    //find a montain peak
    while (i < arr.length - 1 && arr[i] < arr[i + 1]) {
        i++;
    }
    
    if(i === arr.length - 1 || i === 0) {
        return false;
    }
    
    //check if after the peak numbers are decreasing
    for(i; i <= arr.length - 1; i++) {
        if(arr[i] <= arr[i + 1]) {
            return false;
        }
    }
    
    return true;
};

File: 0941-valid-mountain-array/0941-valid-mountain-array.py
class Solution(object):
    def validMountainArray(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        
        if len(arr) < 3:
            return False
        
        i = 0
        
        while i < len(arr) - 1 and arr[i] < arr[i + 1]:
            i += 1
            
        if i == 0 or i == len(arr) - 1:
            return False
        
            
        for x in range(i, len(arr) - 1):
            if arr[x] <= arr[x + 1]:
                return False
            
        return True
        
        